fe/fe-core/src/main/java/com/starrocks/connector/jdbc/OracleSchemaResolver.java

```diff
@@ -0,0 +1,148 @@
+                          String catalogName, Map<String, String> properties) throws DdlException {
+        Map<String, String> newProp = new HashMap<>(properties);
+        newProp.putIfAbsent(JDBCTable.JDBC_TABLENAME, "\"" + dbName + "\"" + "." + "\"" + name + "\"");
+        return new JDBCTable(id, name, schema, partitionColumns, dbName, catalogName, newProp);
+    }
+
+    @Override
+    public List<Column> convertToSRTable(ResultSet columnSet) throws SQLException {
+        List<Column> fullSchema = Lists.newArrayList();
+        while (columnSet.next()) {
+            Type type = convertColumnType(columnSet.getInt("DATA_TYPE"),
+                columnSet.getString("TYPE_NAME"),
+                columnSet.getInt("COLUMN_SIZE"),
+                columnSet.getInt("DECIMAL_DIGITS"));
+            String columnName = columnSet.getString("COLUMN_NAME");
+            fullSchema.add(new Column(columnName, type,
+                columnSet.getString("IS_NULLABLE").equals("YES")));
+        }
+        return fullSchema;
+    }
+
+    @Override
+    public Type convertColumnType(int dataType, String typeName, int columnSize, int digits) {
+        PrimitiveType primitiveType;
+        switch (dataType) {
+            case Types.BIT:
+                primitiveType = PrimitiveType.BOOLEAN;
+                break;
+            case Types.SMALLINT:
+                primitiveType = PrimitiveType.SMALLINT;
+                break;
+            case Types.INTEGER:
+                primitiveType = PrimitiveType.INT;
+                break;
+            case Types.BIGINT:
+            case -1: // LONG
+                primitiveType = PrimitiveType.BIGINT;
+                break;
+            case Types.REAL:
+            case Types.FLOAT:
+            case 100:  // BINARY_FLOAT
+                primitiveType = PrimitiveType.FLOAT;
+                break;
+            case Types.DOUBLE:
+            case 101: // BINARY_DOUBLE
+                primitiveType = PrimitiveType.DOUBLE;
+                break;
+            case Types.NUMERIC:
+            case 3: // NUMBER
+                primitiveType = PrimitiveType.DECIMAL32;
+                break;
+            case Types.CHAR:
+            case Types.NCHAR:
+                return ScalarType.createCharType(columnSize);
+            case Types.VARCHAR:
+            case -9: //NVARCHAR2
+                if (columnSize > 0) {
+                    return ScalarType.createVarcharType(columnSize);
+                } else {
+                    return ScalarType.createVarcharType(ScalarType.CATALOG_MAX_VARCHAR_LENGTH);
+                }
+            case Types.DATE:
+                primitiveType = PrimitiveType.DATE;
+                break;
+            case Types.TIMESTAMP:
+            case -102: // TIMESTAMP WITH LOCAL TIME ZONE
+            case -101: // TIMESTAMP WITH TIME ZONE
+                primitiveType = PrimitiveType.DATETIME;
```

Comment:
for type with time zone, this may lead to result error, could you confirm it?

Reply:
I'll verify the time zone issue, and then I'll add some ut. And i will submit another pull request after verification.

Reply:
> for type with time zone, this may lead to result error, could you confirm it?

I'll verify the time zone issue, and then I'll add some ut. And i will submit another pull request after verification.